diff --git a/includes/commerce_cybersource_sawm.callbacks.inc b/includes/commerce_cybersource_sawm.callbacks.inc
index efb5caa..5f2e726 100644
--- a/includes/commerce_cybersource_sawm.callbacks.inc
+++ b/includes/commerce_cybersource_sawm.callbacks.inc
@@ -505,6 +505,9 @@ function commerce_cybersource_sawm_redirect_form($form, &$form_state, $order, $p
     }
   }
 
+  // Allow other modules an opportunity to add to or alter the transaction data.
+  drupal_alter('commerce_cybersource_sawm_transaction_data', $form, $order);
+
   /*
   All fields should be added to $form by now so they can be signed.
   The signed_field_names is a required field and should list itself.
@@ -534,9 +537,6 @@ function commerce_cybersource_sawm_redirect_form($form, &$form_state, $order, $p
 
   $form['#action'] = $url;
 
-  // Allow other modules an opportunity to add to or alter the transaction data.
-  drupal_alter('commerce_cybersource_sawm_transaction_data', $form);
-
   $form['submit'] = array(
     '#type' => 'submit',
     '#value' => t('Redirect for payment'),
